lib/appmap/trace.rb

```diff
@@ -2,14 +2,36 @@
 
 module AppMap
   module Trace
-    class ScopedMethod < SimpleDelegator
-      attr_reader :package, :defined_class, :static
+    class RubyMethod
```

Comment:
This class is refactored so that a second, duck-typed implementation can be provided.